WordPress/gutenberg
# Framework: Add package-lock precommit check for latest NPM

This pull request seeks to add a pre-commit task which verifies that any changes to the root `package-lock.json` can only be committed if the local NPM version matches the latest available NPM version. This is in line with the [documented required environment](https://github.com/WordPress/gutenberg/blob/master/docs/contributors/getting-started.md), and is intended to resolve cases where `package-lock.json` is wrongly updated with invalid values due to use of outdated version of NPM (see https://github.com/WordPress/gutenberg/issues/16157#issuecomment-601187274).

**Testing Instructions:**

1. Check out the branch
   - `git checkout add/latest-npm-package-lock-check`
2. Install a version of NPM older than the current version
   - `npm i -g npm@6.14.3`
3. Modify `package-lock.json`
   - `echo "example" >> package-lock.json`
4. Add and attempt to commit the change
   - `git add . && git commit -m "test"`
5. Verify that a failure occurs in the precommit and an instructional message is shown
6. Install the latest version of NPM
   - `npm i -g npm@latest`
7. Attempt again to commit the change
   - `git commit -m "test"`
8. Verify that the commit succeeds
